### PR TITLE
ACQ-761 Add base amount getter for PSD2

### PR DESCRIPTION
**Pull request by chivchila**
Add base amount in a data attribute (without currency) for PSD2 requirements.
[ticket](https://financialtimes.atlassian.net/browse/ACQ-761)

### CODE REVIEW

File: components/payment-term.jsx [hunk 1]
```diff
@@ -88,6 +88,11 @@ export function PaymentTerm({
 			),
 		},
 	};
+
+	const createPaymentAmount = (option) => (
+		<input className="ncf__payment-base_amount" type="hidden" value={option.amount}/>
+	);
+
```
Comment (ludrob) [inline components/payment-term.jsx:95]: I've tested an alternative using data attribute and I think it looks much cleaner.
All we need to do is to add the `data-base-amount` to the input props
```
		const props = {
			type: 'radio',
			id: option.value,
			name: inputName,
			value: option.value,
			'data-base-amount': option.amount,
			className:
				'o-forms-input__radio o-forms-input__radio--right ncf__payment-term__input',
			...(option.selected && { defaultChecked: true }),
		};
```

File: utils/payment-term.js [hunk 2]
```diff
@@ -54,6 +55,18 @@ class PaymentTerm {
...
+		if (!checkedDiv) {
+			throw new Error('No payment term has been selected');
+		}
+		const baseElement = checkedDiv.getElementsByClassName(BASE_AMOUNT_CLASS)[0];
+		if (!baseElement) {
+			throw new Error('No base amount element was found');
+		}
+		return baseElement.getAttribute('value');
+	}
+
```
Comment (ludrob) [inline utils/payment-term.js:69]: if we use data-attribute, this simply becomes (dataset is a special key to get the data attributes of an element):
```suggestion
	getBaseAmount () {
		const checked = this.$paymentTerm.querySelector('input:checked');
		if (!checked) {
			throw new Error('No payment term has been selected');
		}
		const baseAmount = checked.dataset.baseAmount;

		return baseAmount;
	}

```

File: components/payment-term.jsx (hunk 1 above)
Comment (chivchila) [reply components/payment-term.jsx]: cleaner idea! thank you!

File: utils/payment-term.js (hunk 2 above)
Comment (chivchila) [reply utils/payment-term.js]: yes, I prefer that solution

File: components/payment-term.jsx [hunk 3]
```diff
@@ -206,6 +207,7 @@ PaymentTerm.propTypes = {
 			selected: PropTypes.bool,
 			trialDuration: PropTypes.string,
 			trialPrice: PropTypes.string,
+			amount: PropTypes.string,
```
Comment (ludrob) [inline components/payment-term.jsx:210]:
```suggestion
			amount: PropTypes.number,
```

File: utils/payment-term.js [hunk 4]
```diff
@@ -54,6 +54,14 @@ class PaymentTerm {
 		return checked.getAttribute('value');
 	}
 
+	getBaseAmount () {
```
Comment (ludrob) [inline utils/payment-term.js:57]:
```suggestion
	/**
	 * Returns the base amount for the selected payment term
	 * @return {Number}
	 * @throws If no payment term has been selected
	 */
	getBaseAmount () {
```

**Review by ludrob — APPROVED** [review]

File: utils/payment-term.js [hunk 5]
```diff
@@ -54,6 +54,19 @@ class PaymentTerm {
 		return checked.getAttribute('value');
 	}
 
+	/**
+	 * Returns the base amount for the selected payment term
+	 * @return {Number}
+	 * @throws If no payment term has been selected
+	 */
+	getBaseAmount () {
```
Comment (umbobabo) [inline utils/payment-term.js:62]: suggestion: You could easily cover this method via test duplicating and change [this test](https://github.com/Financial-Times/n-conversion-forms/blob/75ad2b9ce3795e4c7c5b4fb86be87472341bfd18/test/utils/payment-term.spec.js#L53)

**Review by umbobabo — APPROVED** [review]
Approved, although would be beneficial to implement a simple unit test

File: utils/payment-term.js (hunk 5 above)
Comment (ludrob) [reply utils/payment-term.js:62]: Thank you! Agreed, @chivchila is commiting this exact change right now :)

**Review by ludrob — APPROVED** [review]

**Pull request merged**